Tests in `launcher_parse_args_test.py` fail in VSCode devcontainer

...but only when launched from VSCode environment. They pass when running pytest in terminal. The following assertion fails in `launcher_parse_args_test.py::test_launcher_args_parse_1`:
```python
>       assert path_join(launcher.global_config["pathVarWithEnvVarRef"], abs_path=True) \
            == path_join(expected_root_pwd, "foo", abs_path=True)
E       AssertionError: assert '/workspaces/MlosCore/foo' == '/vscode/vscode-server/bin/linux-x64/d037ac076cee195194f93ce6fe2bdfe2969cc82d/foo'
E         - /vscode/vscode-server/bin/linux-x64/d037ac076cee195194f93ce6fe2bdfe2969cc82d/foo
E         + /workspaces/MlosCore/foo
```

--- a/mlos_bench/mlos_bench/tests/config/globals/global_test_config.jsonc
+++ b/mlos_bench/mlos_bench/tests/config/globals/global_test_config.jsonc
@@ -13,9 +13,13 @@
     "testVnetName": "$testVmName-vnet",
 
     // Note: $PWD generally also works, but has issues in vscode unit test
-    // runner vs. CLI due to the way its wrapper starts in one directory vs.
-    // moves to the code directory without changing the environment variables.
-    "pathVarWithEnvVarRef": "$PWD/foo",
+    // runner vs. CLI due to the way its wrapper sometimes starts in one
+    // directory vs. moves to the code directory without changing the
+    // environment variables.
+    // So, for this reason we set an environment variable explicitly in the
+    // test setup, and ensure that the launcher properly propagates it through
+    // here when expanding variables.
+    "pathVarWithEnvVarRef": "$CUSTOM_PATH_FROM_ENV/foo",
     "varWithEnvVarRef": "user:$USER",
 
     // Override the default value of the "max_iterations" parameter

--- a/mlos_bench/mlos_bench/tests/launcher_parse_args_test.py
+++ b/mlos_bench/mlos_bench/tests/launcher_parse_args_test.py
@@ -52,11 +52,11 @@ def test_launcher_args_parse_1(config_paths: List[str]) -> None:
     """
     # The VSCode pytest wrapper actually starts in a different directory before
     # changing into the code directory, but doesn't update the PWD environment
-    # variable.
-    expected_root_pwd = os.environ.get('VSCODE_CWD', os.getcwd())
+    # variable so we use a separate variable.
+    # See global_test_config.jsonc for more details.
+    os.environ["CUSTOM_PATH_FROM_ENV"] = os.getcwd()
     if sys.platform == 'win32':
         # Some env tweaks for platform compatibility.
-        os.environ['PWD'] = expected_root_pwd
         os.environ['USER'] = os.environ['USERNAME']
 
     # This is part of the minimal required args by the Launcher.
@@ -78,7 +78,7 @@ def test_launcher_args_parse_1(config_paths: List[str]) -> None:
     assert launcher.global_config['test_global_value_2'] == 'from-args'
     # Check that we can expand a $var in a config file that references an environment variable.
     assert path_join(launcher.global_config["pathVarWithEnvVarRef"], abs_path=True) \
-        == path_join(expected_root_pwd, "foo", abs_path=True)
+        == path_join(os.getcwd(), "foo", abs_path=True)
     assert launcher.global_config["varWithEnvVarRef"] == f'user:{getuser()}'
     assert launcher.teardown
     # Check that the environment that got loaded looks to be of the right type.
@@ -97,11 +97,11 @@ def test_launcher_args_parse_2(config_paths: List[str]) -> None:
     """
     # The VSCode pytest wrapper actually starts in a different directory before
     # changing into the code directory, but doesn't update the PWD environment
-    # variable.
-    expected_root_pwd = os.environ.get('VSCODE_CWD', os.getcwd())
+    # variable so we use a separate variable.
+    # See global_test_config.jsonc for more details.
+    os.environ["CUSTOM_PATH_FROM_ENV"] = os.getcwd()
     if sys.platform == 'win32':
         # Some env tweaks for platform compatibility.
-        os.environ['PWD'] = expected_root_pwd
         os.environ['USER'] = os.environ['USERNAME']
 
     config_file = 'cli/test-cli-config.jsonc'
@@ -121,7 +121,7 @@ def test_launcher_args_parse_2(config_paths: List[str]) -> None:
     assert launcher.global_config['testVnetName'] == 'MockeryExperiment-vm-vnet'
     # Check that we can expand a $var in a config file that references an environment variable.
     assert path_join(launcher.global_config["pathVarWithEnvVarRef"], abs_path=True) \
-        == path_join(expected_root_pwd, "foo", abs_path=True)
+        == path_join(os.getcwd(), "foo", abs_path=True)
     assert launcher.global_config["varWithEnvVarRef"] == f'user:{getuser()}'
     assert not launcher.teardown
 

